core/templates/pages/contributor-dashboard-admin-page/translation-role-editor-modal/cd-admin-translation-role-editor-modal.component.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for editing user roles.
+ */
+
+import { Component, OnInit, Input } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+import { ContributorDashboardAdminBackendApiService } from '../services/contributor-dashboard-admin-backend-api.service';
+import { AlertsService } from 'services/alerts.service';
+import constants from 'assets/constants';
+
+@Component({
+  selector: 'cd-admin-translation-role-editor-modal',
+  templateUrl: './cd-admin-translation-role-editor-modal.component.html',
+})
+
+export class CdAdminTranslationRoleEditorModal implements OnInit {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() assignedLanguageIds: string[] = [];
+  @Input() languageIdToName!: {[languageId: string]: string};
+  @Input() username!: string;
+  // Set to null when there is no language left in the list of languages to be
+  // updated. If this value is null, it also means that the 'Add' button
+  // should be disabled.
+  newLanguageId: string | null = null;
+  languageIdInUpdate: string | null = null;
+  languageIdsForSelection: string[] = [];
+
+  constructor(
+    private activeModal: NgbActiveModal,
+    private contributorDashboardAdminBackendApiService:
+      ContributorDashboardAdminBackendApiService,
+    private alertsService: AlertsService
+  ) {}
+
+  private updatelanguageIdsForSelection(): void {
+    this.languageIdsForSelection = Object.keys(this.languageIdToName).filter(
+      languageId => !this.assignedLanguageIds.includes(languageId));
+    this.newLanguageId = this.languageIdsForSelection[0];
```

Comment:
What if `languageIdsForSelection` is empty? Won't this throw index out of bounds?

Reply:
This condition will only occur when we have assigned a user to all the available languages and if this does occur we don't show the option of `addLanguage` in the frontend.

Reply:
![image](https://github.com/oppia/oppia/assets/91470808/a194f442-0d08-43b9-bfa7-278b7d83cc55)